pkg/storegateway/store_gateway_client.go

```diff
@@ -3,7 +3,7 @@
 // Provenance-includes-license: Apache-2.0
 // Provenance-includes-copyright: The Cortex Authors.
 
-package querier
+package storegateway
```

Comment:
WDYT about putting this in `storegateway/client`, same as we do for the ingester client?

Reply:
@56quarters I had a call with @colega in which we discussed that probably the whole implementation of [store_gateway_client.go](https://github.com/grafana/mimir/blob/main/pkg/querier/store_gateway_client.go) (currently belonging to the `querier`, and not `storegateway` package), as well as of [pool.go](https://github.com/grafana/dskit/blob/main/ring/client/pool.go) were done before we started using generics, and therefore the implementation looked a bit more complicated than it should be. The changes I made in `store_gateway_client.go` were needed for a new test that I introduced, and that will now be removed. So, I will also remove the changes I did there. Nevertheless, we find that it might be appropriate to introduce generics in `pool.go` and related sources, and to update mimir accordingly. What do you think about that?